tracer/test/Datadog.Trace.Debugger.IntegrationTests/SignatureParserTests.cs

```diff
@@ -0,0 +1,72 @@
+        [InlineData("ClassWith1Generic`1<T1>,ClassWith2Generics`2<T1,T2>,ClassWith3Generics`3<T1,T2,T3>", new[] { "ClassWith1Generic`1[T1]", "ClassWith2Generics`2[T1,T2]", "ClassWith3Generics`3[T1,T2,T3]" })]
+        [InlineData("A,B,C,D,E,F,G", new[] { "A", "B", "C", "D", "E", "F", "G" })]
+        [InlineData("(A, B, C, D, E, F, G)", new[] { "A", "B", "C", "D", "E", "F", "G" })]
+        [InlineData("(InvalidType<)", null)]
+        [InlineData("InvalidType>)", null)]
+        [InlineData("(InvalidType>)", null)]
+        [InlineData("(A, InvalidType<)", null)]
+        [InlineData("(ValidType<>)", new[] { "ValidType<>" })]
+        [InlineData("WebService.Controllers.MeGenericClass1`1+MeInnerGeneric2`4<WebService.Controllers.MultiAnimal`6<!1,!2,!4,!!0,!!1,WebService.Controllers.Animal`1<!!1>>", null)]
+        [InlineData("Namespace1.Class1`1<Namespace2.Class2`2<Arg1, Arg2>, Namespace3.Class3>", new[] { "Namespace1.Class1`1[Namespace2.Class2`2[Arg1,Arg2],Namespace3.Class3]" })]
+        [InlineData("OuterNamespace.OuterClass`1<InnerNamespace1.InnerClass`2<Arg1,Arg2>,InnerNamespace2.InnerClass>", new[] { "OuterNamespace.OuterClass`1[InnerNamespace1.InnerClass`2[Arg1,Arg2],InnerNamespace2.InnerClass]" })]
+        [InlineData("A.B.C.D`1<A.B.C.E`2<A.B.C.F,A.B.C.G>>", new[] { "A.B.C.D`1[A.B.C.E`2[A.B.C.F,A.B.C.G]]" })]
+        [InlineData("ClassWithGeneric`1<AnotherClassWithGeneric`2<One,Two>,ThirdClass>", new[] { "ClassWithGeneric`1[AnotherClassWithGeneric`2[One,Two],ThirdClass]" })]
+        [InlineData("Namespace1.Class1`1<Namespace2.Class2`1<Namespace3.Class3>>", new[] { "Namespace1.Class1`1[Namespace2.Class2`1[Namespace3.Class3]]" })]
+        [InlineData("(Outer`1<Inner`2<T1,T2>>)", new[] { "Outer`1[Inner`2[T1,T2]]" })]
+        [InlineData("(Outer`1, Inner`2<T1>, T2>", null)]
+        [InlineData("OuterClass<InnerClass1>, A<<InnerClass2>", null)]
+        [InlineData("OuterClass<InnerClass1>, A<InnerClass2>", new[] { "OuterClass[InnerClass1]", "A[InnerClass2]" })]
+        [InlineData("(OuterClass<InnerClass1>, A<<InnerClass2>)", null)]
+        [InlineData("(OuterClass<InnerClass1>, A<InnerClass2>)", new[] { "OuterClass[InnerClass1]", "A[InnerClass2]" })]
+        [InlineData("Namespace1.Class1`1<Namespace2.Class2`1<Namespace3.Class3>", null)]
+        [InlineData("OuterClass<InnerClass1,InnerClass2>", new[] { "OuterClass[InnerClass1,InnerClass2]" })]
+        [InlineData("Class1<Class2<Class3<Class4>>>", new[] { "Class1[Class2[Class3[Class4]]]" })]
+        [InlineData("(Namespace1.Class1`1<Namespace2.Class2`1<Namespace3.Class3>)", null)]
+        [InlineData("Namespace1.Class1`1<Namespace2.<PrivateImplementationDetails>, Namespace3.Class3>", new[] { "Namespace1.Class1`1[Namespace2.[PrivateImplementationDetails],Namespace3.Class3]" })]
+        [InlineData("Namespace1.Class1`1<Namespace2.<>z__ReadOnlyArray<T>, Namespace3.Class3>", new[] { "Namespace1.Class1`1[Namespace2.<>z__ReadOnlyArray[T],Namespace3.Class3]" })]
```

Comment:
I don't think this is actually a valid signature is it? 😅 Probably doesn't matter I guess (and maybe preferable in fact, as shows we're not dependent on parsing the `1

Reply:
Correct, it shall be ``2` as there are two params. I don't parse out the amount of generics and verify that - not sure it's worth it. But in general you're right. If we'll improve the code to actually do that- they will fail 🤓

Reply:
I agree, I don't know that it's worth it either, just wanted to verify. And if we _do_ decide to do that, this will serve as proof it works😃